src/server/bg_services/replication_log_parser.js

```diff
@@ -0,0 +1,176 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const AWS = require('aws-sdk');
+const saw = require('string-saw');
+const moment = require('moment');
+const system_store = require('../system_services/system_store').get_instance();
+const cloud_utils = require('../../util/cloud_utils');
+const http_utils = require('../../util/http_utils');
+const dbg = require('../../util/debug_module')(__filename);
+const config = require('../../../config');
+
+const AWS_ACCESS_LOG_REGEXP = /(?:([a-z0-9]+)|-) (?:-|([a-z0-9.\-_]+)) (?:\[([^\]]+)\]|-) (?:([0-9.]+)|-) (?:-|([a-z0-9:/-]+)) (?:([a-z0-9.-_]+)|-) (?:([a-z.]+)|-) (?:-|([a-z0-9.\-_/%]+)) (?:"-"|"([^"]+)"|-) (?:(\d+)|-) (?:([a-z]+)|-) (?:(\d+)|-) (?:(\d+)|-) (?:(\d+)|-) (?:(\d+)|-) (?:"-"|"([^"]+)"|-) (?:"-"|"([^"]+)"|-) (?:([a-z0-9]+)|-)/i;
+
+async function get_log_candidates(source_bucket_id, replicationconfig, candidates_limit) {
+    if (replicationconfig.aws_log_replication_info) {
+        return get_aws_log_candidates(source_bucket_id, replicationconfig.aws_log_replication_info, candidates_limit);
+    }
+}
+
+async function get_aws_log_candidates(source_bucket_id, aws_log_replication_info, candidates_limit) {
+    const { logs_bucket, logs_prefix, continuation_token } = aws_log_replication_info;
+
+    const s3 = get_source_bucket_aws_connection(source_bucket_id, aws_log_replication_info);
+
+    let candidates_map = {};
+    let logs_retrieved_count = config.AWS_LOG_CANDIDATES_LIMIT;
+
+    do {
+        const next_log_list = await aws_list_next_log(s3, logs_bucket, logs_prefix, continuation_token);
```

Comment:
num of list_objects can be reduced to 1 instead of 10 when using list objects version 1 and passing the key_marker instead of the continuation token as we discussed